### PR TITLE
ci: init

### PR DESCRIPTION
**Pull request by mydoomfr**
* setup Dependabot to update our GitHub Actions and Dockerfile.
* use [fkirc/skip-duplicate-actions](https://github.com/fkirc/skip-duplicate-actions) to save time.
* use [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action).
* use the `make check-diff` job to ensure that `make generate`, `make manifests`, `go mod tidy` return nothing new.
* use `unit-tests` job to run `make test`
* use `sonarqube` job to analyze the code with our SonarQube instance.
* use `publish-artifacts` job to build and push container images to Github registry (ghcr) and perform [Trivy](https://github.com/aquasecurity/trivy-action) analysis.
* add a manual pipeline "Create Release" to tag, create the Github release, generate the changelog based on the semantic commits, and attach the `bundle.yaml` file to the release.

### CODE REVIEW

**Comment by mydoomfr** [comment]
@antrema PR updated and ready for review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        run: go mod download
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Check Diff
+        run: |
+          make check-diff
```
Comment (antrema) [inline .github/workflows/ci.yml:88]: Not defined in Makefile

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          context: .
+          push: true
+          provenance: false
+          platforms: ${{ inputs.build-platform }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: |
+            BUILDTIME=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.created'] }}
+            VERSION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }}
+            REVISION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.revision'] }}
```
Comment (antrema) [inline .github/workflows/build.yml:69]: build-args useless

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mydoomfr) [reply .github/workflows/ci.yml:89]: fixed

File: .github/workflows/build.yml (hunk 2 above)
Comment (mydoomfr) [reply .github/workflows/build.yml]: fixed

**Review by antrema — APPROVED** [review]

**Pull request merged**